src/schnetpack/nn/embed.py

```diff
@@ -0,0 +1,278 @@
+                              operand=r_ij,
+                              fn=lambda y: y / d_ij[:, None],
+                              placeholder=0
+                              )  # shape: (n_pairs, 3)
+        unit_r_ij = safe_scale(unit_r_ij, scale=pair_mask[:, None])  # shape: (n_pairs, 3)
+
+        # Spherical harmonics
+        sph_harms_ij = []
+        for sph_fn in self.sph_fns:
+            sph_ij = safe_scale(sph_fn(unit_r_ij), scale=pair_mask[:, None])  # shape: (n_pairs,2l+1)
+            sph_harms_ij += [sph_ij]  # len: |L| / shape: (n_pairs,2l+1)
+
+        sph_harms_ij = np.concatenate(sph_harms_ij, axis=-1) if len(self.degrees) > 0 else None
+        # shape: (n_pairs,m_tot)
+
+        geometric_data = {'R': R,
+                          'r_ij': r_ij,
+                          'unit_r_ij': unit_r_ij,
+                          'd_ij': d_ij,
+                          'rbf_ij': rbf_ij,
+                          'phi_r_cut': phi_r_cut,
+                          'sph_ij': sph_harms_ij,
+                          }
+
+        # Spherical harmonic coordinates (SPHCs)
+        if self.sphc:
+            z = inputs[self.atomic_type_key]
+            point_mask = inputs['point_mask']
+            if self.sphc_normalization is None:
+                # Initialize SPHCs to zero
+                geometric_data.update(_init_sphc_zeros(z=z,
+                                                       sph_ij=sph_harms_ij,
+                                                       phi_r_cut=phi_r_cut,
+                                                       idx_i=idx_i,
+                                                       point_mask=point_mask,
+                                                       mp_normalization=self._lambda)
+                                      )
+            else:
+                # Initialize SPHCs with a neighborhood dependent embedding
+                geometric_data.update(_init_sphc(z=z,
+                                                 sph_ij=sph_harms_ij,
+                                                 phi_r_cut=phi_r_cut,
+                                                 idx_i=idx_i,
+                                                 point_mask=point_mask,
+                                                 mp_normalization=self._lambda)
+                                      )
+
+        # Solid harmonics (Spherical harmonics + radial part)
+        if self.solid_harmonic:
+            rbf_ij = safe_scale(rbf_ij, scale=phi_r_cut[:, None])  # shape: (n_pairs,K)
+            g_ij = sph_harms_ij[:, :, None] * rbf_ij[:, None, :]  # shape: (n_pair,m_tot,K)
+            g_ij = safe_scale(g_ij, scale=pair_mask[:, None, None], placeholder=0)  # shape: (n_pair,m_tot,K)
+            geometric_data.update({'g_ij': g_ij})
+
+        return geometric_data
+
+
+class AtomTypeEmbed(nn.Module):
+    def __init__(self, features: int, prop_keys: Dict, num_embeddings: int = 100,
```

Comment:
Atomtype embedding is implemented in [here](https://github.com/atomistic-machine-learning/schnetpack/blob/337fa3f803725201b5890ec4ee07da0d4774dec8/src/schnetpack/nn/embedding.py#L152) or directly in the init of the [representation](https://github.com/atomistic-machine-learning/schnetpack/blob/337fa3f803725201b5890ec4ee07da0d4774dec8/src/schnetpack/representation/schnet.py#L133-L134)